test/client-stream.js

```diff
@@ -0,0 +1,42 @@
+'use strict'
+
+const { test } = require('tap')
+const { Client } = require('..')
+const { createServer } = require('http')
+const { PassThrough } = require('stream')
+
+test('basic get w stream', (t) => {
+  t.plan(7)
+
+  const server = createServer((req, res) => {
+    t.strictEqual('/', req.url)
+    t.strictEqual('GET', req.method)
+    t.strictEqual('localhost', req.headers.host)
+    res.setHeader('content-type', 'text/plain')
+    res.end('hello')
+  })
+  t.tearDown(server.close.bind(server))
+
+  server.listen(0, () => {
+    const client = new Client(`http://localhost:${server.address().port}`)
+    t.tearDown(client.close.bind(client))
+
+    client.stream({
+      path: '/',
+      method: 'GET'
+    }, (err, { statusCode, headers }) => {
+      const pt = new PassThrough()
+      t.error(err)
+      t.strictEqual(statusCode, 200)
+      t.strictEqual(headers['content-type'], 'text/plain')
+      const bufs = []
+      pt.on('data', (buf) => {
+        bufs.push(buf)
+      })
+      pt.on('end', () => {
+        t.strictEqual('hello', Buffer.concat(bufs).toString('utf8'))
+      })
+      return pt
+    })
+  })
```

Comment:
Can you please add a few tests for the errors?

Reply:
added more tests